scripts/signature.py

```diff
@@ -0,0 +1,122 @@
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import json
+import sys
+
+import pyvelox.pyvelox as pv
+from deepdiff import DeepDiff
+
+
+# Utility to export and diff function signatures.
+
+
+# From https://stackoverflow.com/questions/287871/how-do-i-print-colored-text-to-the-terminal
+class bcolors:
+    OKCYAN = "\033[96m"
+    OKGREEN = "\033[92m"
+    WARNING = "\033[93m"
+    FAIL = "\033[91m"
+    BOLD = "\033[1m"
+
+
+def export(args):
+    """Exports Velox function signatures."""
+    if args.spark:
+        pv.register_spark_signatures()
+    else:
+        pv.register_presto_signatures()
+
+    signatures = pv.get_function_signatures()
+
+    # Convert signatures to json
+    jsoned_signatures = {}
+    for key in signatures.keys():
+        jsoned_signatures[key] = [str(value) for value in signatures[key]]
+
+    # Persist to file
+    json.dump(jsoned_signatures, args.output_file)
+
+
+def diff(args):
+    """Diffs Velox function signatures."""
+    first_signatures = json.load(args.first)
+    second_signatures = json.load(args.second)
+    delta = DeepDiff(
+        first_signatures, second_signatures, ignore_order=True, report_repetition=True
+    )
+    if delta:
+        if "dictionary_item_removed" in delta:
+            # Exit gracefully if a function is removed
+            print(
+                f"Signature removed: {bcolors.FAIL}{delta['dictionary_item_removed']}"
+            )
+            sys.exit(1)
+
+        if "values_changed" in delta:
+            # Exit gracefully if values in a function signature are changed
+            print(f"Signature changed: {bcolors.FAIL}{delta['values_changed']}")
+            sys.exit(1)
+
+        if "repetition_change" in delta:
+            # Exit gracefully if a function signature is repeated
+            print(f"Signature repeated: {bcolors.FAIL}{delta['repetition_change']}")
+            sys.exit(1)
+
+        print(f"Found differences: {bcolors.OKGREEN}{delta}")
+    else:
+        print(f"{bcolors.BOLD}No differences found.")
+
+
+def help(args):
+    parser.print_help()
+    return 0
+
+
+def parse_args():
+    global parser
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description="""Velox Function Signature Utility
+Usage: signature.py [OPTIONS] COMMAND [ARGS]...
+
+Commands:
+  diff    Diffs Velox function signatures.
+  export  Exports Velox function signatures.
+                
+""",
+    )
+
+    command = parser.add_subparsers(dest="command")
+    export_command_parser = command.add_parser("export")
+    export_command_parser.add_argument("--spark", action="store_true")
```

Comment:
for consistency, should have a --presto flag that's True by default? (what if you want to export both)

Reply:
Sg , added presto flag.